charts/helm-chart/kubernetes-dashboard/values.yaml

```diff
@@ -12,288 +12,286 @@
-  ## If not set, the global containterSecurityContext values will define these values
-  # containerSecurityContext:
-  #   allowPrivilegeEscalation: false
-  #   readOnlyRootFilesystem: true
-  #   runAsUser: 1001
-  #   runAsGroup: 2001
-#  args:
-#    - --log-level=info
-#    - --logtostderr=true
+  containers:
+    ports:
+      - name: metrics-scraper
+        containerPort: 8000
+        protocol: TCP
+    args: []
+    # Additional container environment variables
+    # env:
+    # - name: SOME_VAR
+    #   value: 'some value'
+    env: []
+    # Additional volume mounts
+    # - mountPath: /kubeconfig
+    #   name: dashboard-kubeconfig
+    #   readOnly: true
+    volumeMounts:
+      # Create volume mount to store logs (required)
+      - mountPath: /tmp
+        name: tmp-volume
+    # TODO: Validate configuration
+    resources:
+      requests:
+        cpu: 100m
+        memory: 200Mi
+      limits:
+        cpu: 250m
+        memory: 400Mi
+    livenessProbe:
+      httpGet:
+        scheme: HTTP
+        path: /
+        port: 8000
+      initialDelaySeconds: 30
+      timeoutSeconds: 30
+  # Additional volumes
+  # - name: dashboard-kubeconfig
+  #   secret:
+  #     defaultMode: 420
+  #     secretName: dashboard-kubeconfig
+  volumes:
+    # Create on-disk volume to store exec logs (required)
+    - name: tmp-volume
+      emptyDir: {}
+  nodeSelector:
+    # TODO: check if it's really needed since we offer cross platform images for darwin/windows/linux
+    kubernetes.io/os: linux
+  # Labels & annotations shared between WEB UI related resources
+  labels: {}
+  annotations: {}
+
 
-## Optional Metrics Server sub-chart
+## Optional Metrics Server sub-chart configuration
 ## Enable this if you don't already have metrics-server enabled on your cluster and
 ## want to use it with dashboard metrics-scraper
 ## refs:
 ##  - https://github.com/kubernetes-sigs/metrics-server
 ##  - https://github.com/kubernetes-sigs/metrics-server/tree/master/charts/metrics-server
 metrics-server:
-  enabled: false
-  ## Example for additional args
-  # args:
-  #  - --kubelet-preferred-address-types=InternalIP
-  #  - --kubelet-insecure-tls
-
-rbac:
-  # Specifies whether namespaced RBAC resources (Role, Rolebinding) should be created
-  create: true
-
-  # Specifies whether cluster-wide RBAC resources (ClusterRole, ClusterRolebinding) to access metrics should be created
-  # Independent from rbac.create parameter.
-  clusterRoleMetrics: true
-
+  enabled: true
+  args:
+    - --kubelet-preferred-address-types=InternalIP
+    - --kubelet-insecure-tls
+
+## Optional Cert Manager sub-chart configuration
+## Enable this if you don't already have cert-manager enabled on your cluster.
+cert-manager:
+  enabled: true
+  installCRDs: true
+
+## Optional Nginx Ingress sub-chart configuration
+## Enable this if you don't already have nginx-ingress enabled on your cluster.
+nginx:
+  enabled: true
```

Comment:
I wonder if nginx & cert-manager should be enabled by default: those are great demo values, but in production almost everybody will manage those two.

Would having sane but safe defaults for production in values.yaml, and having another "demo" values file on top of that, acceptable?

Reply:
We always wanted to provide a fully working installation by default. Technically even if the user already has both installed it would still work for both groups of users. On the other hand, disabling them by default won't work for users without those dependencies installed.

If there is a simple way to provide some default templates on install that could be selected and used by the users, it would be best. I am not that familiar with helm though.